salt/_pillar/metalk8s_nodes.py

```diff
@@ -50,13 +50,103 @@ def node_info(node, ca_minion):
     return result
 
 
+def get_cluster_version(api_client):
+
+    coreV1 = kubernetes.client.CoreV1Api(api_client=api_client)
+    namespace = coreV1.read_namespace(name="kube-system")
+    annotation_key = 'metalk8s.scality.com/cluster-version'
+
+    if annotation_key not in namespace.metadata.annotations:
+        return __utils__['pillar_utils.errors_to_dict'](
+            'Unable to retrieve cluster version'
+        )
+
+    return namespace.metadata.annotations[annotation_key]
+
+
+def list_storage_classes(api_client):
+
+    storageV1 = kubernetes.client.StorageV1Api(api_client=api_client)
+    storage_classes = {}
+
+    storageclass_list = storageV1.list_storage_class()
+    if not storageclass_list:
+        return {}
+    for storageclass in storageclass_list.items:
+        storage_classes[storageclass.metadata.name] = storageclass
+
+    return storage_classes
+
+
+def list_volumes(api_client, minion_id):
+
+    customObjectApi = kubernetes.client.CustomObjectsApi(api_client=api_client)
+    volume_information = {}
+
+    storage_classes = list_storage_classes(api_client)
+
+    try:
+        volume_list = customObjectApi.list_cluster_custom_object(
+            group="storage.metalk8s.scality.com",
+            version="v1alpha1",
+            plural="volumes"
+        )
+
+        if not volume_list:
+            volume_information = __utils__['pillar_utils.errors_to_dict']([
+                'Unable to retrieve node volumes'
+            ])
+
+        storage_classes = list_storage_classes(api_client)
+
+        if not storage_classes:
+            return __utils__['pillar_utils.errors_to_dict'](
+                ['Unable to retrieve node volume list']
+            )
+
+        for items in volume_list.get('items'):
+            if items['spec']['nodeName'] != minion_id:
+                continue
+            storageclass = storage_classes.get(
+                items['spec']['storageClassName']
+            )
+            if storageclass is None:
+                volume_information = __utils__['pillar_utils.errors_to_dict']([
+                    'No known storage class found for {}'.format(
+                        items['metadata']['name']
+                    )
+                ])
+            else:
+                result_vol_object = {
+                    'name': items['metadata']['name'],
+                    'uid': items['metadata']['uid'],
+                    'capacity': items.get('spec').get(
+                        'sparseLoopDevice', {}).get('size', ""),
+                    'storage_class': items.get('spec', {}).get(
+                        'storageClassName', ""),
+                    'status': items.get('status', {}).get('phase', ""),
+                    'storageclass_annotations':
+                        storageclass.metadata.annotations
```

Comment:
See my other comment: we won't use annotations but `parameters` ;)

Reply:
And really, don't pass it in like this. We're making things more complicated, less future-proof etc than it could be.